### PR TITLE
Add Settings app

### PR DESCRIPTION
**Pull request by PanSpagetka**
Add Settings app and move sources and RBAC into it.

### CODE REVIEW

File: main.yml [hunk 1]
```diff
@@ -352,8 +350,17 @@ settings:
   title: Cloud Services Settings
   frontend:
     sub_apps:
+      - id: ''
+        default: true
+        title: 'User Preferences'
```
Comment (ryelo) [inline main.yml]: I think the mocks had this as "General"

https://marvelapp.com/5a4j976/screen/63049476

File: main.yml [hunk 2]
```diff
@@ -352,8 +350,17 @@ settings:
   title: Cloud Services Settings
```
Comment (ryelo) [inline main.yml]: This should just be "Settings"

https://marvelapp.com/5a4j976/screen/63049476

File: main.yml [hunk 3]
```diff
@@ -352,8 +350,17 @@ settings:
...
+        default: true
+        title: 'User Preferences'
       - id: hooks
+      - id: sources
+      - id: rbac
   top_level: true
+  isBeta: true
+  deployment_repo: https://github.com/RedHatInsights/settings-frontend-build
+  paths:
+  - /settings
```
Comment (ryelo) [inline main.yml]: Let's indent this

**Review by ryelo — CHANGES_REQUESTED** [review]
Just going off the mocks, a couple little naming things.

I'll sync with Kendra to double check

File: main.yml [hunk 4]
```diff
@@ -352,8 +350,17 @@ settings:
...
 
 ruledev:
```
Comment (kruai) [inline main.yml:375]: I'm not sure why the `settings` app is above `ruledev`, but I can fix that after merge.

File: main.yml (hunk 3 above)
Comment (kruai) [reply main.yml]: To clarify, just the `- /settings` line needs to be indented here.

**Review by kruai — CHANGES_REQUESTED** [review]
I agree with Ryan's comments.

File: main.yml (hunk 3 above)
Comment (karelhala) [inline main.yml]: There's going to be another path in newar future for settings app `/settings/app` I don't really know how to structure the ID here @kruai any ideas. The url has to be `/settings/${smething}` and the `something` should indicate that it's a app settings. ID with `app` sounds strange and someone reading this file could understand it as placeholder.

Comment (karelhala) [reply main.yml]: We'll need another `- settings/${config}` path, where I don't really know how to name the `config` app. We'll be configurating application in tat section, so `app` would be nice, but that's not really a good application ID. @kruai any ideas?

Comment (PanSpagetka) [reply main.yml]: We can call it for example `- settings/sites` so it the whole url will look like `settings/sites/insights`. What do you think?

Comment (ryelo) [reply main.yml]: Let's wait to see what Kendra's mocks say, but before it was going to be 
`settings/applications/insights` (but I'm sure apps would be fine)

File: main.yml [hunk 5]
```diff
@@ -349,11 +347,32 @@ rhel:
...
+        default: true
+        title: 'General'
       - id: hooks
+      - id: sources
+      - id: rbac
+      - id: sites
   top_level: true
+  isBeta: true
+  deployment_repo: https://github.com/RedHatInsights/settings-frontend-build
+  paths:
```
Comment (kruai) [inline main.yml]: I also just noticed that `paths` isn't under `frontend` (it should be)

File: main.yml [hunk 6]
```diff
@@ -349,11 +347,32 @@ rhel:
...
+
+sites:
+  title: Applications
+  frontend:
+    sub_apps:
+      - id: insights
+        title: Insights
+  isBeta: true
+  deployment_repo: https://github.com/RedHatInsights/settings-frontend-build
+  paths:
```
Comment (kruai) [inline main.yml]: I also just noticed that `paths` isn't under `frontend` (it should be)

**Comment by PanSpagetka** [comment]
@kruai @ryelo @karelhala Please re-review, indentation should be right and paths should match https://github.com/RedHatInsights/settings-frontend/pull/10  if there are request for different paths/titles we can figure it out in next PR.

**Review by kruai — APPROVED** [review]
Having 2 applications define the same deployment repos will waste some space and CPU cycles, but we can figure out how to refactor a bit later. Otherwise, I think this looks good.

**Review by karelhala — APPROVED** [review]

File: main.yml [hunk 7]
```diff
@@ -349,11 +347,32 @@ rhel:
...
+      - /settings/applications
+
+applications:
+  title: Applications
+  isBeta: true
+  deployment_repo: https://github.com/RedHatInsights/settings-frontend-build
+  frontend:
+    sub_apps:
+      - id: insights
+          title: Insights
```
Comment (ryelo) [inline main.yml]: This line is too far indented

**Review by ryelo — CHANGES_REQUESTED** [review]

File: main.yml (hunk 7 above)
Comment (PanSpagetka) [reply main.yml]: Damn...fixed!

File: main.yml [hunk 8]
```diff
@@ -349,11 +347,32 @@ rhel:
   top_level: true
 
 settings:
-  title: Cloud Services Settings
+  title: Settings
+  top_level: true
+  isBeta: true
```
Comment (kruai) [inline main.yml]: Sorry I noticed this a little late, but `isBeta` is a field that only exists under `api`. Check out how it's used in other apps, like Approval, for instance.

File: main.yml [hunk 9]
```diff
@@ -349,11 +347,32 @@ rhel:
...
-  top_level: true
+      - id: sources
+      - id: rbac
+      - id: applications
+    paths:
+      - /settings/applications
+
+applications:
+  title: Applications
+  isBeta: true
```
Comment (kruai) [inline main.yml]: Same issue here.

**Review by kruai — CHANGES_REQUESTED** [review]
Sorry, but I noticed another field that's out of place!

**Review by kruai — APPROVED** [review]
Looks good now.

**Review by ryelo — APPROVED** [review]

**Pull request merged**